samples/outTreeEp/out_tree_ep.cc

```diff
@@ -0,0 +1,105 @@
+#include "out_tree_ep.h"
+#include <memory>
+#include <vector>
+namespace onnxruntime {
+
+OutTreeEp::OutTreeEp(const char* ep_type, const OutTreeEpInfo& ep_info) : info(ep_info) {
+    type = ep_type;
+    OrtExecutionProvider::GetCapability = [](const OrtExecutionProvider* this_, const OrtGraphViewer* graph, size_t* cnt, OrtIndexedSubGraph*** indexed_sub_graph) {
```

Comment:
If I'm understanding correctly, the type of the `OrtIndexedSubGraph*** indexed_sub_graph` parameter is essentially asking the EP to fill out an array of _pointers_ to `OrtIndexedSubGraph` objects.

Would it be simpler to change this to `OrtIndexedSubgraph** indexed_sub_graph` so that the EP fills out an array of `OrtIndexedSubGraph` objects directly? Each OrtIndexedSubgraph struct is a simple POD that can be created on the stack and copied around. It seems like it would result in less pointer tracking.

Reply:
Also, who is responsible for freeing this memory and when? If the EP allocates an array, then the EP should free it. The currently example leaks the allocations.

Reply:
The problem is that we don't know how may OrtIndexedSubGraph would be before we call GetCapability() function. I will fix the leak issue in the coming commits